symbolic-symcache/src/lib.rs

```diff
@@ -168,116 +168,82 @@ impl<'data> SymCache<'data> {
+            .ok_or(ErrorKind::InvalidRanges)?;
 
-        if buf.len() < expected_buf_size || source_locations_size < ranges_size {
-            return Err(ErrorKind::BadFormatLength.into());
+        let (_, rest) = align_to(rest, 8).ok_or(ErrorKind::UnexpectedStringBytes {
+            expected: header.string_bytes as usize,
+            found: 0,
+        })?;
+        if rest.len() < header.string_bytes as usize {
+            return Err(ErrorKind::UnexpectedStringBytes {
+                expected: header.string_bytes as usize,
+                found: rest.len(),
+            }
+            .into());
         }
 
-        // SAFETY: we just made sure that all the pointers we are constructing via pointer
-        // arithmetic are within `buf`
-        let files_start = unsafe { buf.as_ptr().add(header_size) };
-        let functions_start = unsafe { files_start.add(files_size) };
-        let source_locations_start = unsafe { functions_start.add(functions_size) };
-        let ranges_start = unsafe { source_locations_start.add(source_locations_size) };
-        let string_bytes_start = unsafe { ranges_start.add(ranges_size) };
-
-        // SAFETY: the above buffer size check also made sure we are not going out of bounds
-        // here
-        let files = unsafe {
-            &*ptr::slice_from_raw_parts(files_start as *const raw::File, header.num_files as usize)
-        };
-        let functions = unsafe {
-            &*ptr::slice_from_raw_parts(
-                functions_start as *const raw::Function,
-                header.num_functions as usize,
-            )
-        };
-        let source_locations = unsafe {
-            &*ptr::slice_from_raw_parts(
-                source_locations_start as *const raw::SourceLocation,
-                header.num_source_locations as usize,
-            )
-        };
-        let ranges = unsafe {
-            &*ptr::slice_from_raw_parts(
-                ranges_start as *const raw::Range,
-                header.num_ranges as usize,
-            )
-        };
-        let string_bytes = unsafe {
-            &*ptr::slice_from_raw_parts(string_bytes_start, header.string_bytes as usize)
-        };
-
         Ok(SymCache {
             header,
             files,
             functions,
             source_locations,
             ranges,
-            string_bytes,
+            string_bytes: rest,
         })
     }
 
     /// Resolves a string reference to the pointed-to `&str` data.
     fn get_string(&self, offset: u32) -> Option<&'data str> {
-        if offset == u32::MAX {
-            return None;
-        }
-        let len_offset = offset as usize;
-        let len_size = std::mem::size_of::<u32>();
-        let len = u32::from_ne_bytes(
-            self.string_bytes
-                .get(len_offset..len_offset + len_size)?
-                .try_into()
-                .unwrap(),
-        ) as usize;
+        if self.header.version == SYMCACHE_VERSION {
```

Comment:
```suggestion
        if self.header.version >= 8 {
```

this matches the comment and is forward compatible to newer versions.